drivers/net/phy/sfp.c

```diff
@@ -337,6 +337,23 @@ static unsigned int sff_gpio_get_state(struct sfp *sfp)
 	return sfp_gpio_get_state(sfp) | SFP_F_PRESENT;
 }
 
+/**
+ * simulate_pull_up - simulate a pull-up on a GPIO
+ * @desc: GPIO descriptor
+ * 
+ * Simulate a pull-up on a GPIO by driving it low for a short time and then
+ * driving it high. This is used to work around eratta in the Siklu N366 PCB
+ * where the TX_DISABLE GPIO is not pulled up. 
+ * 
+ * As a result, the SFP module may not be able to detect the change between 
+ * TX_DISABLE=1 to TX_DISABLE=0, and the other side will be see "link down".
+ */
+static void simulate_pull_up(struct gpio_desc *desc) {
+	gpiod_direction_output(desc, 0);
+	udelay(10);
+	gpiod_direction_output(desc, 1);
```

Comment:
we can set the `GPIO_TX_DISABLE` output with the level of 1 to disable the TX
but maybe we don't need to set it to 0 and then to 1 after 10usec - we only need to do it once.
What about the following code?
```
        static int is_initialized = 0;
        if (is_initialized == 0) {
                 initialized = 1;
	         gpiod_direction_output(desc, 0);
	         udelay(10);
        }
	gpiod_direction_output(desc, 1);
```

Reply:
no, we need to do it every time the SFP disconnects and reconnects again. not only once.

Reply:
Maybe you are right. enabling TX for 10usec, is not critical. We can stay with it.